### DISCUSSION TITLE
想问一下初始化如何选择设备？

### DISCUSSION THREAD

**Discussion by desktop4025** [Mrs4s/go-cqhttp#2134] [Q&A]
试了mirai只有macos成功登录过，但现在mirai怎么样都登不上了，所以想在生成设备时选择一下设备类型，但我不知道在哪里改。。config里面好像没有找到选项

**Comment by yyx2684264** [comment]
不在config里面，启动gocq后会生成一个device.json，其中的"protocol": 6, 的值就是设备类型，具体参考docs中每种类型对应的值:[https://docs.go-cqhttp.org/guide/config.html#%E9%85%8D%E7%BD%AE%E4%BF%A1%E6%81%AF](url)

**Comment by desktop4025** [comment]
好的谢谢，最近搞了半天终于用mirai的扫码登录登上了
